lib/Sema/TypeCheckError.cpp

```diff
@@ -1647,6 +1674,280 @@ class CheckErrorCoverage : public ErrorHandlingWalker<CheckErrorCoverage> {
+    auto isSubscript = isa<SubscriptExpr>(E->getSubExpr());
+    auto expr = isSubscript ? E->getSubExpr() : E;
+
+    if (!isa<AssignExpr>(expr) || !isa<MemberRefExpr>(expr) ||
+        !isa<SubscriptExpr>(expr)) {
+      return ShouldRecurse;
+    }
+
+    return checkPotentiallyThrowingAccessorThrowSite(expr, /*isSetter*/ false,
+                                                     isSubscript);
+  }
+
+  ShouldRecurse_t checkSubscriptExpr(SubscriptExpr *E) {
+    return checkPotentiallyThrowingAccessorThrowSite(E, /*isSetter*/ false,
+                                                     /*isSubscript*/ true);
+  }
+
+  ShouldRecurse_t checkPotentiallyThrowingAccessorThrowSite(Expr *E,
+                                                            bool isSetter,
+                                                            bool isSubscript) {
+    ContextScope scope(*this, None);
+    auto walker = ThrowingAccessorWalker(E, &Flags, TC, /*tryLoc*/ SourceLoc(),
+                                         /*handled*/ false);
+    auto throws = walker.checkAccessor(E, isSetter, isSubscript);
+
+    if (throws) {
+      auto classification = Classification(ThrowingKind::Throws,
+                                           PotentialReason::forThrowingApply());
+      checkThrowSite(E, /*requiresTry*/ true, classification);
+    }
+
+    return ShouldNotRecurse;
+  }
+
+  void checkPotentiallyThrowingAccessor(Expr *E, ContextScope &scope,
+                                        TypeChecker &typeChecker,
+                                        SourceLoc tryLoc) {
+    auto isTryHandled = isa<OptionalTryExpr>(E) || isa<ForceTryExpr>(E);
+    auto walker = ThrowingAccessorWalker(E, &Flags, typeChecker, tryLoc,
+                                         /*handled*/ isTryHandled);
+    E->walk(walker);
+  }
+
+  class ThrowingAccessorWalker : public ASTWalker {
+  private:
+    Expr *baseExpr;
+    ContextFlags *contextFlags;
+    TypeChecker &typeChecker;
+    SourceLoc tryLoc;
+    bool isTryHandled;
+
+    std::pair<bool, Expr *> walkToExprPre(Expr *E) override {
+      if (isa<AssignExpr>(E)) {
+        auto AE = cast<AssignExpr>(E);
+        if (auto SE = dyn_cast<SubscriptExpr>(AE->getDest())) {
+          return {checkSubscript(SE, contextFlags, typeChecker, tryLoc,
+                                 isTryHandled, true),
+                  nullptr};
+        }
+        if (auto OEE = dyn_cast<OpenExistentialExpr>(AE->getDest())) {
+          if (isa<SubscriptExpr>(OEE->getSubExpr())) {
+            return {checkSubscript(cast<SubscriptExpr>(OEE->getSubExpr()),
+                                   contextFlags, typeChecker, tryLoc,
+                                   isTryHandled, true),
+                    nullptr};
+          }
+        }
+
+        return {checkComputedPropertySetter(AE, contextFlags, typeChecker,
+                                            tryLoc, isTryHandled),
+                nullptr};
+      }
```

Comment:
Yeah, we really need to reintroduce `LValueAccessKind`.  That can be a separate PR.

Reply:
@rjmccall Do we need it to get this PR merged or can it be added later? I have a PR ready to bring back `LValueAccessKind` but it's been a long time since it was removed, so unfortunately the PR breaks a bunch of stuff. Maybe @slavapestov could take a look at bringing it back properly since he removed it?

Reply:
I think it's reasonable to ask Slava to help with that, yes.  But I do think we should block this PR on that one instead of adding new logic like this; there's enough of this sort of thing as it is.

Reply:
@rjmccall Okay! I read Slava's comment on SR-9960 about bringing back LValueAccessKind as some sort of reusable [AST] visitor. If Slava has some free some, then I'll let him do it, otherwise I can pick up the work if someone could point me towards how the computation is supposed to work.

Reply:
or... I can put up my PR that reverts the LValueAccessKind code that @slavapestov deleted and we can go from there - i.e. fix whatever is broken.